drafts/migration-to-python-3-only.md

```diff
@@ -0,0 +1,228 @@
+# Migration to Python 3 only
+
+This document discuss some of the various possibilities to migrate Jupyter, and
+its reference Kernel implementation IPython, toward a Python 3 only code base,
+without breaking installation for Python 2 users.
+
+The document is not meant to debate whether the Jupyter & IPython project should migrate
+to a Python 3 only code base, but the timeline of such a migration.
+
+
+# TL;DR:
```

Comment:
Could you change this header to something more readable? I always forget what it means :-)